pkg/util/yaml_test.go

```diff
@@ -233,3 +234,25 @@ func TestToYAMLWithJSONPB(t *testing.T) {
 		})
 	}
 }
+
+func TestUnmarshalWithJSONPB(t *testing.T) {
+	unmarshalWithJSONPBTests := []struct {
+		desc string
+		yaml string
+		want *v1alpha2.IstioControlPlaneSpec
+	}{
+		{"UnmarshalWithJSONPBToYAML", icpYaml, icp},
+	}
+
+	for _, tt := range unmarshalWithJSONPBTests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got := &v1alpha2.IstioControlPlaneSpec{}
+			err := UnmarshalWithJSONPB(icpYaml, got)
+			if err != nil {
```

Comment:
should have err!=nil check before? then only compare when err is nil

Reply:
done.

Reply:
s/t.Errorf/t.Fatalf unless you want the test case to continue after unexpected errors.